tests/System.Buffers.Primitives.Tests/BufferReferenceTesting.cs

```diff
@@ -106,15 +106,12 @@ static void Buffer(OwnedBuffer<byte> buffer)
         static void Pin(OwnedBuffer<byte> buffer)
         {
             var memory = buffer.Buffer;
-            Assert.False(buffer.IsRetained);
```

Comment:
Are you talking about this particular test or others as well, like MemoryHandleDoubleFree?

Should I remove the Pin test? It is still testing that pinning returns a non-zero pointer address and that Dispose doesn't throw.